src/spatial_ref/srs.rs

```diff
@@ -451,6 +451,141 @@ impl SpatialRef {
             None
         }
     }
+
+    /// Get spheroid semi-major axis.
+    ///
+    /// Returns [`Err`] variant in case the C library returns an error code different from [`OGRERR_NONE`](https://gdal.org/api/vector_c_api.html#c.OGRERR_NONE).
```

Comment:
To be honest, I'm not really a fan of these, since they basically say "the function returns `Err` if the C function returned an error", which is true, but useless.

But I don't mind them too much, so it's all right if you like them.

Reply:
I maybe tried to be to precise and I created to explicit documentation.

Reply:
Docs are good, but this is currently neither consistent with the rest of the documentation, nor useful: "the function returns `Err` if an error has occurred" doesn't bring any information at all to someone used to the Rust conventions.

Looking at `GetSemiMajor`, it's a bit weird -- it returns both a valid value (`SRS_WGS84_SEMIMAJOR`) and an error if it can't find the value, which is something I'm not sure we want to emulate.

Can you change this to something like "Returns an error if the semi-major axis can't be found" ("semi-minor" for the other one), "if there the `PROJCS` node is missing" for `set_proj_param`?

`GetProjParm` returns an error if it can't find `PROJCS` or `CONVERSION`, or if the parameter is missing. `OSRSetAttrValue` returns an error if the path is empty. `OSRGetAttrValue` returns an error if the attribute can't be found.

For `geog_cs`, it's a bit complicated, I guess an error means that it's missing.

Also, `OSRSetAttrValue` in GDAL can take a null value, which means that the node is created without a value. Maybe we should take an `Option<&str>` to support that?

Reply:
Docs are good, but this is currently neither consistent with the rest of the documentation, nor useful: "the function returns `Err` if an error has occurred" doesn't bring any information at all to someone used to the Rust conventions.

Looking at `GetSemiMajor`, it's a bit weird -- it returns both a valid value (`SRS_WGS84_SEMIMAJOR`) and an error if it can't find the value, which is something I'm not sure we want to emulate.

Can you change this to something like "Returns an error if the semi-major axis can't be found" ("semi-minor" for the other one), "if there the `PROJCS` node is missing" for `set_proj_param`?

`GetProjParm` returns an error if it can't find `PROJCS` or `CONVERSION`, or if the parameter is missing. `OSRSetAttrValue` returns an error if the path is empty. `OSRGetAttrValue` returns an error if the attribute can't be found.

For `geog_cs`, it's a bit complicated, I guess an error means that it's missing.

These expose some of the GDAL internals, but I don't expect them to change significantly.

Also, `OSRSetAttrValue` in GDAL can take a null value, which means that the node is created without a value. Maybe we should take an `Option<&str>` to support that?

Reply:
@lnicola:
I've corrected comments.

I've also decided to change signature of `get_proj_param` to:
```rust
pub fn get_proj_param(&self, name: &str) -> Result<Option<f64>>
```
I did not include a default value option, because in each case this value will be discarded.

I've done similar thing to `get_attr_value`.

There is also problem with `set_attr_value`. I've tried to write some tests which set some silly (like "Spam" or "Eggs") and less silly (like "WGS_1984" and "metre") data, but results were quite strange. I'm not an expert at manipulation of WKT for projection, so I maybe tried to write tests in wrong way. Maybe some of You will have better idea how to write those tests.

Alternatively we can remove this function now and implement it later.

Reply:
> but results were quite strange

You're right, I'll try to look into it.

Reply:
I took the liberty of pushing a commit to fix this. It was a pretty basic gotcha -- `Option::map` consumes the value. I would have hoped that clippy would catch this, but alas...

Reply:
Thanks. Rust/C++ edge can be tricky.

Reply:
Everything can be tricky. The Python bindings will segfault if you do something similar with a SpatialRef, so it's not just us.